### DISCUSSION TITLE
Installation on Fedora issue

### DISCUSSION THREAD

**Discussion by mzigaib** [comfyanonymous/ComfyUI#5668] [Q&A]
I am using Fedora 41 and I am trying to install it but I got stuck on the installing requirements part, I was wondering if someone could help me.
Here is my log:

Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: torch in /home/mzigaib/.local/lib/python3.13/site-packages (from -r requirements.txt (line 1)) (2.6.0.dev20241118+cu124)
Collecting torchsde (from -r requirements.txt (line 2))
  Using cached torchsde-0.2.6-py3-none-any.whl.metadata (5.3 kB)
Requirement already satisfied: torchvision in /home/mzigaib/.local/lib/python3.13/site-packages (from -r requirements.txt (line 3)) (0.20.0.dev20241118+cu124)
Requirement already satisfied: torchaudio in /home/mzigaib/.local/lib/python3.13/site-packages (from -r requirements.txt (line 4)) (2.5.0.dev20241118+cu124)
Collecting einops (from -r requirements.txt (line 5))
  Using cached einops-0.8.0-py3-none-any.whl.metadata (12 kB)
Collecting transformers>=4.28.1 (from -r requirements.txt (line 6))
  Using cached transformers-4.46.3-py3-none-any.whl.metadata (44 kB)
Collecting tokenizers>=0.13.3 (from -r requirements.txt (line 7))
  Using cached tokenizers-0.20.3-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (6.7 kB)
Collecting sentencepiece (from -r requirements.txt (line 8))
  Using cached sentencepiece-0.2.0.tar.gz (2.6 MB)
  Preparing metadata (setup.py) ... done
Collecting safetensors>=0.4.2 (from -r requirements.txt (line 9))
  Using cached safetensors-0.4.5-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (3.8 kB)
Collecting aiohttp (from -r requirements.txt (line 10))
  Using cached aiohttp-3.11.2-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (7.7 kB)
Requirement already satisfied: pyyaml in /usr/lib64/python3.13/site-packages (from -r requirements.txt (line 11)) (6.0.1)
Requirement already satisfied: Pillow in /usr/lib64/python3.13/site-packages (from -r requirements.txt (line 12)) (11.0.0)
Collecting scipy (from -r requirements.txt (line 13))
  Using cached scipy-1.14.1-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (60 kB)
Requirement already satisfied: tqdm in /usr/local/lib/python3.13/site-packages (from -r requirements.txt (line 14)) (4.67.0)
Collecting psutil (from -r requirements.txt (line 15))
  Using cached psutil-6.1.0-cp36-abi3-manylinux_2_12_x86_64.manylinux2010_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (22 kB)
Collecting kornia>=0.7.1 (from -r requirements.txt (line 18))
  Using cached kornia-0.7.4-py2.py3-none-any.whl.metadata (18 kB)
Collecting spandrel (from -r requirements.txt (line 19))
  Using cached spandrel-0.4.0-py3-none-any.whl.metadata (15 kB)
Collecting soundfile (from -r requirements.txt (line 20))
  Using cached soundfile-0.12.1-py2.py3-none-manylinux_2_31_x86_64.whl.metadata (14 kB)
Requirement already satisfied: filelock in /home/mzigaib/.local/lib/python3.13/site-packages (from torch->-r requirements.txt (line 1)) (3.16.1)
Requirement already satisfied: typing-extensions>=4.10.0 in /usr/lib/python3.13/site-packages (from torch->-r requirements.txt (line 1)) (4.12.2)
Requirement already satisfied: networkx in /home/mzigaib/.local/lib/python3.13/site-packages (from torch->-r requirements.txt (line 1)) (3.4.2)
Requirement already satisfied: jinja2 in /home/mzigaib/.local/lib/python3.13/site-packages (from torch->-r requirements.txt (line 1)) (3.1.4)
Requirement already satisfied: fsspec in /home/mzigaib/.local/lib/python3.13/site-packages (from torch->-r requirements.txt (line 1)) (2024.10.0)
Requirement already satisfied: nvidia-cuda-nvrtc-cu12==12.4.127 in /home/mzigaib/.local/lib/python3.13/site-packages (from torch->-r requirements.txt (line 1)) (12.4.127)
Requirement already satisfied: nvidia-cuda-runtime-cu12==12.4.127 in /home/mzigaib/.local/lib/python3.13/site-packages (from torch->-r requirements.txt (line 1)) (12.4.127)
Requirement already satisfied: nvidia-cuda-cupti-cu12==12.4.127 in /home/mzigaib/.local/lib/python3.13/site-packages (from torch->-r requirements.txt (line 1)) (12.4.127)
Requirement already satisfied: nvidia-cudnn-cu12==9.1.0.70 in /home/mzigaib/.local/lib/python3.13/site-packages (from torch->-r requirements.txt (line 1)) (9.1.0.70)
Requirement already satisfied: nvidia-cublas-cu12==12.4.5.8 in /home/mzigaib/.local/lib/python3.13/site-packages (from torch->-r requirements.txt (line 1)) (12.4.5.8)
Requirement already satisfied: nvidia-cufft-cu12==11.2.1.3 in /home/mzigaib/.local/lib/python3.13/site-packages (from torch->-r requirements.txt (line 1)) (11.2.1.3)
Requirement already satisfied: nvidia-curand-cu12==10.3.5.147 in /home/mzigaib/.local/lib/python3.13/site-packages (from torch->-r requirements.txt (line 1)) (10.3.5.147)
Requirement already satisfied: nvidia-cusolver-cu12==11.6.1.9 in /home/mzigaib/.local/lib/python3.13/site-packages (from torch->-r requirements.txt (line 1)) (11.6.1.9)
Requirement already satisfied: nvidia-cusparse-cu12==12.3.1.170 in /home/mzigaib/.local/lib/python3.13/site-packages (from torch->-r requirements.txt (line 1)) (12.3.1.170)
Requirement already satisfied: nvidia-cusparselt-cu12==0.6.2 in /home/mzigaib/.local/lib/python3.13/site-packages (from torch->-r requirements.txt (line 1)) (0.6.2)
Requirement already satisfied: nvidia-nccl-cu12==2.21.5 in /home/mzigaib/.local/lib/python3.13/site-packages (from torch->-r requirements.txt (line 1)) (2.21.5)
Requirement already satisfied: nvidia-nvtx-cu12==12.4.127 in /home/mzigaib/.local/lib/python3.13/site-packages (from torch->-r requirements.txt (line 1)) (12.4.127)
Requirement already satisfied: nvidia-nvjitlink-cu12==12.4.127 in /home/mzigaib/.local/lib/python3.13/site-packages (from torch->-r requirements.txt (line 1)) (12.4.127)
Requirement already satisfied: setuptools in /home/mzigaib/.local/lib/python3.13/site-packages (from torch->-r requirements.txt (line 1)) (75.5.0)
Requirement already satisfied: sympy==1.13.1 in /home/mzigaib/.local/lib/python3.13/site-packages (from torch->-r requirements.txt (line 1)) (1.13.1)
Requirement already satisfied: mpmath<1.4,>=1.1.0 in /home/mzigaib/.local/lib/python3.13/site-packages (from sympy==1.13.1->torch->-r requirements.txt (line 1)) (1.3.0)
Requirement already satisfied: numpy>=1.19 in /usr/local/lib64/python3.13/site-packages (from torchsde->-r requirements.txt (line 2)) (2.1.3)
Collecting trampoline>=0.1.2 (from torchsde->-r requirements.txt (line 2))
  Using cached trampoline-0.1.2-py3-none-any.whl.metadata (10 kB)
Collecting huggingface-hub<1.0,>=0.23.2 (from transformers>=4.28.1->-r requirements.txt (line 6))
  Using cached huggingface_hub-0.26.2-py3-none-any.whl.metadata (13 kB)
Requirement already satisfied: packaging>=20.0 in /usr/lib/python3.13/site-packages (from transformers>=4.28.1->-r requirements.txt (line 6)) (24.1)
Requirement already satisfied: regex!=2019.12.17 in /usr/lib64/python3.13/site-packages (from transformers>=4.28.1->-r requirements.txt (line 6)) (2024.9.11)
Requirement already satisfied: requests in /usr/lib/python3.13/site-packages (from transformers>=4.28.1->-r requirements.txt (line 6)) (2.32.3)
Collecting aiohappyeyeballs>=2.3.0 (from aiohttp->-r requirements.txt (line 10))
  Using cached aiohappyeyeballs-2.4.3-py3-none-any.whl.metadata (6.1 kB)
Collecting aiosignal>=1.1.2 (from aiohttp->-r requirements.txt (line 10))
  Using cached aiosignal-1.3.1-py3-none-any.whl.metadata (4.0 kB)
Collecting attrs>=17.3.0 (from aiohttp->-r requirements.txt (line 10))
  Using cached attrs-24.2.0-py3-none-any.whl.metadata (11 kB)
Collecting frozenlist>=1.1.1 (from aiohttp->-r requirements.txt (line 10))
  Using cached frozenlist-1.5.0-cp313-cp313-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (13 kB)
Collecting multidict<7.0,>=4.5 (from aiohttp->-r requirements.txt (line 10))
  Using cached multidict-6.1.0-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (5.0 kB)
Collecting propcache>=0.2.0 (from aiohttp->-r requirements.txt (line 10))
  Using cached propcache-0.2.0-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (7.7 kB)
Collecting yarl<2.0,>=1.17.0 (from aiohttp->-r requirements.txt (line 10))
  Using cached yarl-1.17.2-cp313-cp313-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (66 kB)
Collecting kornia-rs>=0.1.0 (from kornia>=0.7.1->-r requirements.txt (line 18))
  Using cached kornia_rs-0.1.7.tar.gz (71 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [10 lines of output]
      error: failed to parse manifest at `/tmp/pip-install-ru_tudog/kornia-rs_9475ada19fd14109a9b99fb079d211b9/kornia-py/Cargo.toml`
      
      Caused by:
        can't find library `kornia_rs`, rename file to `src/lib.rs` or specify lib.path
      💥 maturin failed
        Caused by: Cargo metadata failed. Does your crate compile with `cargo build`?
        Caused by: `cargo metadata` exited with an error:
      Error running maturin: Command '['maturin', 'pep517', 'write-dist-info', '--metadata-directory', '/tmp/pip-modern-metadata-h7508s9l', '--interpreter', '/usr/bin/python3']' returned non-zero exit status 1.
      Checking for Rust toolchain....
      Running `maturin pep517 write-dist-info --metadata-directory /tmp/pip-modern-metadata-h7508s9l --interpreter /usr/bin/python3`
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

Thanks!

**Comment by ltdrdata** [comment]
Don't install to system python.
You need to create venv for python.
And install dependencies and launch ComfyUI inside activated venv.

```
python3 -m venv venv
. venv/bin/activate
```

I'm not sure about Fedora, but in the case of Ubuntu, it was necessary to install the venv package to use venv.

**Comment by comfyanonymous** [comment]
Fedora uses python 3.13 by default, you need to `dnf install python3.12` and then: `python3.12 -m ensurepip` to install pip, then you can: `python3.12 -m pip install ...` torch and the requirements.

**Comment by mzigaib** [comment]
Thank you very much for the feedback! I will try that, I will let you know if it works.

**Comment by mzigaib** [comment]
It totally worked!
Thanks guys!

**Comment by mzigaib** [comment]
Sorry guy but now inside ComfyUI I am getting this error

[START] Security scan
[DONE] Security scan
## ComfyUI-Manager: installing dependencies. (GitPython)
/usr/bin/python3.12: No module named pip
## [ERROR] ComfyUI-Manager: Attempting to reinstall dependencies using an alternative method.
/usr/bin/python3.12: No module named pip
## [ERROR] ComfyUI-Manager: Failed to install the GitPython package in the correct Python environment. Please install it manually in the appropriate environment. (You can seek help at https://app.element.io/#/room/%23comfyui_space%3Amatrix.org)
## [ERROR] ComfyUI-Manager: GitPython package seems to be installed, but failed to load somehow. Make sure you have a working git client installed
** ComfyUI startup time: 2024-11-18 23:26:12.593470
** Platform: Linux
** Python version: 3.12.7 (main, Oct  1 2024, 00:00:00) [GCC 14.2.1 20240912 (Red Hat 14.2.1-3)]
** Python executable: /usr/bin/python3.12
** ComfyUI Path: /opt/ComfyUI
** Log path: /opt/ComfyUI/comfyui.log

Prestartup times for custom nodes:
   0.4 seconds: /opt/ComfyUI/custom_nodes/ComfyUI-Manager

Total VRAM 15967 MB, total RAM 128708 MB
pytorch version: 2.5.1+cu124
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA GeForce RTX 4070 Ti SUPER : cudaMallocAsync
Using pytorch cross attention
[Prompt Server] web root: /opt/ComfyUI/web
Traceback (most recent call last):
  File "/opt/ComfyUI/nodes.py", line 2022, in load_custom_node
    module_spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/opt/ComfyUI/custom_nodes/ComfyUI-Manager/__init__.py", line 6, in <module>
    from .glob import manager_server
  File "/opt/ComfyUI/custom_nodes/ComfyUI-Manager/glob/manager_server.py", line 13, in <module>
    import git
ModuleNotFoundError: No module named 'git'

Cannot import /opt/ComfyUI/custom_nodes/ComfyUI-Manager module for custom nodes: No module named 'git'

Import times for custom nodes:
   0.0 seconds: /opt/ComfyUI/custom_nodes/websocket_image_save.py
   0.0 seconds (IMPORT FAILED): /opt/ComfyUI/custom_nodes/ComfyUI-Manager

Starting server

To see the GUI go to: http://127.0.0.1:8188

**Comment by mzigaib** [comment]
My UI is loading like this, Is that correct?
![image](https://github.com/user-attachments/assets/ebec361d-1fd9-4672-9240-1f0fff7f0530)

**Comment by mzigaib** [comment]
Oh I see now, that error is when I try to install Comfy Manager
Any tips?
